x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph/layout_graph.ts

```diff
@@ -18,26 +30,41 @@ export const layoutGraph = (
   const nodesById: { [key: string]: Node<NodeViewModel> } = {};
   const graphOpts = {
     compound: true,
+    directed: true,
   };
 
   const g = new Dagre.graphlib.Graph(graphOpts)
-    .setGraph({ rankdir: 'LR', align: 'UL' })
+    .setGraph({
+      rankdir: 'LR',
+      align: 'UL',
+      ranksep: GRID_SIZE_OFFSET * 3,
+    })
     .setDefaultEdgeLabel(() => ({}));
 
   edges.forEach((edge) => g.setEdge(edge.source, edge.target));
+  const nodesOfParent: { [key: string]: Array<Node<NodeViewModel>> } = {};
 
+  nodes.forEach((node) => {
+    if (node.parentId) {
+      nodesOfParent[node.parentId] = nodesOfParent[node.parentId] || [];
+      nodesOfParent[node.parentId].push(node);
+    }
+  });
   nodes.forEach((node) => {
     let size = { width: NODE_WIDTH, height: node.measured?.height ?? NODE_HEIGHT };
 
     if (node.data.shape === 'label') {
-      size = calcLabelSize(node.data.label);
+      size = {
+        height: NODE_LABEL_HEIGHT,
+        width: NODE_LABEL_WIDTH,
+      };
 
       // TODO: waiting for a fix: https://github.com/dagrejs/dagre/issues/238
       // if (node.parentId) {
       //   g.setParent(node.id, node.parentId);
       // }
     } else if (node.data.shape === 'group') {
-      const res = layoutGroupChildren(node, nodes);
+      const res = layoutStackedLabels(node, nodesOfParent[node.id]);
```

Comment:
overused `group` term (with the future work of grouped nodes and labels).
Stacked is the right fit in this use case